Python/UpeachyaKansakar/tasks/task1/store_result.py

```diff
@@ -0,0 +1,67 @@
+import os
+import csv
+import argparse
+import datetime as dt
+
+POSSIBLE_SUBJECTS = ["english", "nepali", "mathematics", "science"]
+
+def isValidDate (date):
+    try:
+        dt.datetime.strptime(date, '%Y/%m/%d')
+        return True
+    except ValueError: 
+        return False
+
+def getValidRecord(args):
+    if isValidDate(args.dob):
+        return [args.name, args.dob, args.subject, args.score, args.total]
+    else:
+        raise Exception(args.dob+" is not a valid date. Please enter in yyyy/mm/dd format")
+
+def writeToFile(file, args, isFirstRecord):
+    record_adder = csv.writer(file, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
+    try:
+        # adding header row in case of new file
+        if isFirstRecord:
+            record_adder.writerow(["name", "dob", "subject", "score", "total"])
+        record_adder.writerow(getValidRecord(args))
+    except ValueError as error:
+        print("ERROR: ",error)
+
+def isDuplicateRecord (args):
+    with open(args.store, mode='r') as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        for row in csv_reader:
+            if row["name"] == args.name and row["subject"].lower() == args.subject.lower():
+                return True
+    csv_file.close()
```

Comment:
No need to close the file. Closing of the file is being handled by the context manager

Reply:
:ok: